_episodes/03-index-slice-subset.md

```diff
@@ -358,22 +358,28 @@ gives the **output**
 Remember that Python indexing begins at 0. So, the index location [2, 6]
 selects the element that is 3 rows down and 7 columns over in the DataFrame.
 
+It is worth noting that rows are selected when using `loc` with a single list of labels (or `iloc` with a single list of integers). However, unlike `loc` or `iloc`, indexing a data frame directly with labels will select columns, while ranges of integers will select rows. Direct indexing of rows is redundant with using `iloc`, and will raise a `KeyError` if a single integer or list is used; the error will also occur if index labels are used without `loc` (or column labels used with it).
+A useful rule of thumb is the following: integer-based slicing is best done with `iloc` and will avoid errors (and is generally consistent with indexing of Numpy arrays), label-based slicing of rows is done with `loc`, and slicing of columns by directly indexing column names.
 
 
 > ## Challenge - Range
 >
 > 1. What happens when you execute:
 >
 >    - `surveys_df[0:1]`
+>    - `surveys_df[0]`
 >    - `surveys_df[:4]`
 >    - `surveys_df[:-1]`
 >
 > 2. What happens when you call:
 >
+>    - `surveys_df.iloc[0:1]`
+>    - `surveys_df.iloc[0]`
+>    - `surveys_df.iloc[:4, :]`
 >    - `surveys_df.iloc[0:4, 1:4]`
 >    - `surveys_df.loc[0:4, 1:4]`
 >
-> - How are the two commands different?
+> - How are the last two commands different?
```

Comment:
Unfortunately I do not have the context for these changes to the exercise. Furthermore, I would like to see any changes to challenges accompanied by updates to the discussions of/solutions to those challenges, which are currently kept in the [instructor notes](https://github.com/datacarpentry/python-ecology-lesson/blob/gh-pages/_extras/guide.md) for this lesson.

Reply:
I think these changes can be useful and updated the instructor notes to my best abilities. I also included the examples you suggested for the text above. If @caesoma doesn't mind, we can merge it like that.

Reply:
Thank you. Sorry I didn't answer right away. It's been a while indeed, so I'd have to revise the entire lesson to see how the proposal from then would fit. If @LilithElina or anybody else got around to reviewing it, I don't have any issue with it being incorporated as you see fit.

Reply:
Thank you. I'm sorry I took over like that unasked, I suddenly had free time on my hands and I'm not sure when my next free time slot will be.

Reply:
Totally fine by me. I'm in the opposite situation right now. Not sure when I'll have free time for it, so take it away.